include/scip2/walltime.h

```diff
@@ -38,13 +40,34 @@ class Walltime
       initialized_ = true;
     }
 
-    if (time_device < (1 << DEVICE_TIMESTAMP_BITS) / 2 &&
-        (1 << DEVICE_TIMESTAMP_BITS) / 2 < time_device_prev_)
+    if (detectDeviceTimeJump(time_device))
+    {
+      logger::warn() << "Device time jumped." << std::endl;
```

Comment:
This line seems better to be after L50.

And it would be better to have another time jump check (with more small threshold value) to the places just after calling `Walltime.update()`. (it may be a different PR.)

Reply:
Yes, `Walltime.update()` just dedicates to convert device time to walltime. 
Time jump check with more small threshold value will be added in another PR.